### PR TITLE
Update project and CI

### PR DESCRIPTION
**Pull request by marcelveldt**
Housekeeping PR as the repo was missing a lot of linting and CI and was a bit behind in standards.

- Bump python version to 3.10 to use newer style type annotations
- Move to pyproject.toml
- Add basic linting configuration using ruff
- Add pre-commit
- Update existing code so it will work with the updated lint config
- Add missing EffectStatus entries

Also switched to automatic version numbering following our other libs.

Commented out pylint and mypy in the precommit config, will fix those in follow up PR(s) as it needs a bit more work on the code and this PR would be huge otherwise.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          with open("pyproject.toml", "wb") as f:
+            tomli_w.dump(pyproject, f)
+      - name: Build
+        run: >-
+          python3 -m build
+      - name: Publish release to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.8.10
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```
Comment (edenhaus) [inline .github/workflows/release.yml:45]: We could directly switch to [Trusted Publisher](https://docs.pypi.org/trusted-publishers/) to improve security

File: aiohue/v1/api.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 from __future__ import annotations
 
+# ruff: noqa: D100,D101,D102,D105,D107,PLR0913
```
Comment (edenhaus) [inline aiohue/v1/api.py:3]: I suggest excluding these rules in the general ruff for now.

File: aiohue/v2/models/feature.py [hunk 3]
```diff
@@ -296,10 +296,13 @@ class EffectStatus(Enum):
     CANDLE = "candle"
     FIRE = "fire"
     SPARKLE = "sparkle"
+    GLISTEN = "glisten"
+    OPAL = "opal"
+    PRISM = "prism"
```
Comment (edenhaus) [inline aiohue/v2/models/feature.py:301]: Do you think these additions should be part of this PR?

File: aiohue/v1/api.py (hunk 2 above)
Comment (marcelveldt) [reply aiohue/v1/api.py:3]: Well, I do want to make a general exception.
The v1 code is legacy code we do not touch anymore (hence these exceptions). The v2 code should be properly linted and up to date with code standards

File: .github/workflows/release.yml (hunk 1 above)
Comment (marcelveldt) [reply .github/workflows/release.yml:45]: Ah yes, can we do that in a follow up? I'm not sure how to do that with the libs org and I think the PYPI account linked is still from @balloob

File: aiohue/v2/models/feature.py (hunk 3 above)
Comment (marcelveldt) [reply aiohue/v2/models/feature.py:301]: Nah, but it was too much work to create a new PR just for these ;-)
I spotted these by accident

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+show-fixes = true
+
+# enable later: "C90", "PTH", "TCH", "RET", "ANN"
+select = ["E", "F", "W", "I", "N", "D", "UP", "PL", "Q", "SIM", "TID", "ARG"]
+ignore = ["PLR2004", "N818"]
+extend-exclude = ["app_vars.py"]
+unfixable = ["F841"]
+line-length = 100
+target-version = "py311"
+
```
Comment (edenhaus) [inline pyproject.toml:97]:
```suggestion

[tool.ruff.per-file-ignores]
"aiohue/v1/*" = ["D100", "D101", "D102", "D105", "D107", "PLR0913"]

```

File: aiohue/v1/api.py (hunk 2 above)
Comment (edenhaus) [reply aiohue/v1/api.py:3]: See my suggestion on `pyproject.toml` below

**Review by edenhaus — APPROVED** [review]

**Pull request merged**